src/2d/shallow/surge/explicit_storm_module.f90

```diff
@@ -0,0 +1,506 @@
+
+    ! ==========================================================================
+    !  read_explicit_storm_data()
+    !    Reads storm fields for next time snapshot
+    !    Currently only for ASCII files
+    ! ==========================================================================
+
+    subroutine read_explicit_storm(storm,t)
+
+        implicit none
+
+        ! Subroutine I/O
+        type(explicit_storm_type), intent(in out) :: storm
+        real(kind=8), intent(in) :: t
+
+        ! Local storage
+        integer :: j, k
+        character(len=100) :: data_path
+
+        ! Reading buffer variables
+        integer :: timestamp
+
+        ! Overwrite older storm states with newer storm states
+        storm%t_prev = storm%t_next
+        storm%u_prev = storm%u_next 
+        storm%v_prev = storm%v_next 
+        storm%p_prev = storm%p_next 
+        
+        ! Current time t currently unused in favor of storm%last_storm_index.
+        ! This should probably be changed in the future.
+
+        ! Read the u-velocity file
+        !
+        data_path = trim(storm%data_path_root) // "u10.dat"
+        call read_explicit_storm_file(data_path,storm%u_next,storm%num_rows,storm%last_storm_index,timestamp)
+
+        ! Read v-velocity file
+        !
+        data_path = trim(storm%data_path_root) // "v10.dat"
+        call read_explicit_storm_file(data_path,storm%v_next,storm%num_rows,storm%last_storm_index,timestamp)
+
+        ! Read pressure file
+        !
+        data_path = trim(storm%data_path_root) // "pmsl.dat"
+        call read_explicit_storm_file(data_path,storm%p_next,storm%num_rows,storm%last_storm_index,timestamp)
+
+        ! Convert pressure units: mbar to Pa
+        storm%p_next = storm%p_next * 1.0e2
+
+        ! Save timestamp (sec) of next snapshot
+        storm%t_next = timestamp
+
+        ! Update number of storm snapshots read in
+        storm%last_storm_index = storm%last_storm_index + 1
+        if (DEBUG) print *, "last_storm_index=", storm%last_storm_index
+
+
+    end subroutine read_explicit_storm
+
+    ! ==========================================================================
+    !  integer pure get_lat_index(lat)
+    !    Returns index of latitude array of the storm data
+    !    corresponding to input lat.
+    ! ==========================================================================
+    pure integer function get_lat_index(lat,storm) result(i)
+      
+        implicit none
+
+        ! Input
+        real(kind=8), intent(in) :: lat
+        type(explicit_storm_type), intent(in) :: storm
+
+        ! Since latitude data isn't regularly spaced,
+        !  iterate through until we find the first
+        !  smaller value
+        do i=1,storm%num_rows-1
+            !if (storm%lat(i) < lat) exit
+            if ((storm%lat(i+1) > lat ) .and. (lat > storm%lat(i))) exit
+        enddo
+        ! TODO improve index search; add failure condition
+
```

Comment:
Also if we send a failure condition we should set the field to the ambient pressure and 0 velocity.  If this ends up leading to too much of a discontinuity in the field then we should come up with a smooth ramping function at the edges.